src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs

```diff
@@ -292,55 +295,86 @@
+                           (map (fn [k] {:header k :starting-width attr-col-width :sort-by :text
+                                         :as-text
+                                         (fn [attr-value]
+                                           (cond
+                                             (is-single-ref? attr-value) (attr-value "entityName")
+                                             (is-ref-list? attr-value) (map #(% "entityName") attr-value)
+                                             :else attr-value))
+                                         :content-renderer
+                                         (fn [attr-value]
+                                           (cond
+                                             (is-single-ref? attr-value) (attr-value "entityName")
+                                             (is-ref-list? attr-value) (join ", " (map #(% "entityName") attr-value))
+                                             :else ((:attribute-renderer props) attr-value)))})
+                             attribute-keys))
+                :filters (mapv (fn [[type count]] {:text type :count count :pred (constantly true)})
+                           entity-types)
+                :selected-filter-index (max 0 (.indexOf (to-array (map first entity-types))
+                                                selected-entity-type))
+                :on-filter-change (fn [index]
+                                    (let [type (first (nth entity-types index))]
+                                      (swap! state update-in [:server-response] assoc :selected-entity-type type)
+                                      (react/call :load-type this type)
+                                      (when-let [func (:on-filter-change props)]
+                                        (func type))))
+                :data entity-list
+                :->row (fn [m]
+                         (into [m] (map (fn [k] (get-in m ["attributes" k])) attribute-keys)))})]))]))
+   :component-did-mount
+   (fn [{:keys [props state this]}]
+     (endpoints/call-ajax-orch
+       {:endpoint (endpoints/get-entity-types (:workspace-id props))
+        :on-done (fn [{:keys [success? get-parsed-response]}]
+                   (if success?
+                     (let [types (vec (get-parsed-response))
+                           first-type (or (:initial-entity-type props) (first (first types)))]
+                       (swap! state update-in [:server-response]
+                         assoc :entity-types types :selected-entity-type first-type)
+                       (react/call :load-type this first-type))
+                     (swap! state update-in [:server-response]
+                       assoc :server-error (get-parsed-response))))}))
+   :load-type
+   (fn [{:keys [props state]} type]
+     (if-not type
+       (swap! state update-in [:server-response] assoc :entity-list [])
+       (do
+         (swap! state update-in [:server-response] dissoc :entity-list)
```

Comment:
Let's let the user continue to look at their existing data while the new data is loading.